### ISSUE TITLE
Check for incorrect GitHub Actions if statements

### ISSUE THREAD

**Issue by tunetheweb** [adrienverge/yamllint#405]
It would be good if yamllinyt could detect the issue I described here: https://github.community/t/github-actions-if-statements-can-easily-be-coded-incorrectly/200621

Repeated here for reference:

> I've noticed some repos using this syntax:
> 
> ```yml
>     if: ${{ github.event_name }} == 'schedule'
> ```
> 
> They think it means this:
> 
> ```yml
>     if: ${{ github.event_name == 'schedule' }}
> ```
> 
> But it does not because only the first part is an expression evaluation, so it always returns true (there is always a non-empty `event_name`) so the subsequent statements are always run, **even if this is not a schedule run.**!
> 
> This can be more worrying if workflows are checking branch names (e.g. so certain access happens when a protected `main` or `master` branch is changed), or checking users (so only certain users can run an action) as these may not be adding the protection that users think it is.
> 
> It would be better if GitHub Actions either recognised this syntax error and highlighted it to the user, or evaluated the whole expression as developer no doubt intended, rather than silently implementing the incorrect logic.

**Comment by adrienverge** [comment]
Hello,

It seems to me that this is not related to YAML, but to the specific syntax used by another project (GitHub Actions). Is that right?

**Comment by tunetheweb** [comment]
Yes that is correct. So understand how you might consider it out of scope for this, but I (and others) do use yamllint to lint our GitHub Actions workflow YAML files. I don’t believe there is a dedicated GitHub Actions YAML linter.

It could perhaps be an optional that is only checked if the file is in the `.github/workflows` directory?

**Comment by adrienverge** [comment]
Thank you for the clarification!

I understand the use-case, however yamllint is dedicated to general-purpose YAML, it is not designed to lint specificities of GitHub Actions, of Ansible, of Symfony, or any other third-party project.

In my opinion, if there is a need for GitHub Action syntax, then someone should write a linter or just a script to achieve that.

**Comment by tunetheweb** [comment]
Fair enough and understandable. Closing.
